mu/virtual_environment.py

```diff
@@ -822,6 +824,25 @@ def install_jupyter_kernel(self):
                 "Unable to install kernel\n%s" % compact(output)
             )
 
+    def install_from_zipped_wheels(self, zipped_wheels_filepath):
+        with tempfile.TemporaryDirectory() as unpacked_wheels_dirpath:
+            #
+            # The wheel files are shipped in Mu-version-specific zip files to avoid
+            # cross-contamination when a Mu version change happens and we still have
+            # wheels from the previous installation.
+            #
+            with zipfile.ZipFile(zipped_wheels_filepath) as zip:
+                zip.extractall(unpacked_wheels_dirpath)
+
+            self.reset_pip()
+            for wheel in unpacked_wheels_dirpath:

```

Comment:
Isn't `unpacked_wheels_dirpath` a string to the path where the wheels are unzipped?
If so, woudn't this loop iterate through the string characters?

Reply:
/me laughs. Well that certainly explains the weird test failure I was getting. (I pushed the code up in a hurry knowing there were test failures, but assuming they were odd artefacts of the refactor).

Thanks for that catch. Just put a fix in.